### PR TITLE
Feature/download from cached documents

### PR DESCRIPTION
**Pull request by THOR300**
### Download From Cached Documents

Facilitate the downloading of documents from the cached s3 bucket rather than from source. 

Only persist the original source url in the parser input as a top level field (the download url will be entered into the metadata field). 

(The cached download url isn't needed in the parser as it downloads the document from the cdn)

Via integration tests I have confirmed that: 

- The download_url is used to download the document if both source_url and download_url are present. 
- The download_url is preserved in the metadata field. 
- The source_url is always preserved in the top level field for document_source_url in the parser_input. 

I've also updated some source urls in the test data that referenced climate laws. 

NOTE: The files changed looks more than it is due to the timestamp on the archived documents in the integration tests changing. This could be fixed in future.

### CODE REVIEW

File: integration_tests/data/pipeline_out/ingest_unit_test_parser_input/TESTCCLW.executive.1332.1553.json [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
   "document_id": "TESTCCLW.executive.1332.1553",
   "document_source_url": "https://www.lse.ac.uk/GranthamInstitute/wp-content/uploads/laws/1674%20English.pdf",
```
Comment (THOR300) [inline integration_tests/data/pipeline_out/ingest_unit_test_parser_input/TESTCCLW.executive.1332.1553.json:5]: This source url now redirects to the cclw homepage

File: src/navigator_data_ingest/base/types.py [hunk 2]
```diff
@@ -80,6 +80,7 @@ class Document(BaseModel):
     slug: str  # Unique identifier created by backend
     publication_ts: datetime
     source_url: Optional[AnyHttpUrl]  # Original source URL
+    download_url: Optional[AnyHttpUrl]  # Cached document URL
```
Comment (joel-wright) [inline src/navigator_data_ingest/base/types.py:83]: will this actually be an HTTP URL? I thought we were either supplying an s3 URL or object path?

**Review by joel-wright — APPROVED** [review]
The change looks good, but where did all the tests data changes come from? And were they also required?

File: src/navigator_data_ingest/base/types.py (hunk 2 above)
Comment (THOR300) [reply src/navigator_data_ingest/base/types.py:83]: Seeing as we use requests to download the document my assumption was that we would pass in an s3 object url or cdn domain / path to object to facilitate download. 

"download_url": "https://cpr-data-bucket.s3.eu-west-1.amazonaws.com/prefix/TESTCCLW.executive.2.2.json"

or 

"download_url": "https://cdn.dev.climatepolicyradar.org/prefix/TESTCCLW.executive.2.2.json"

**Pull request merged**